### ISSUE TITLE
Clarification of batch job error

### ISSUE THREAD

**Issue by soxofaan** [Open-EO/openeo-api#436]
In aggregator's large area processing feature I'm currently assuming a batch job error is not recoverable (so once status is "error" , it will never change anymore).

Or am I misinterpreting? For example: can a batch job be restarted (e.g. go back to status "queued"/"running") after it reached status "error"? Or  can/must it be canceled first to do that?

**Comment by soxofaan** [comment]
(I guess we should have a batch job status transition diagram somewhere in the docs)

**Comment by m-mohr** [comment]
This is mostly up to the back-end to decide, but in principle, a batch job with status error, finished or canceled can be started/queued again. What happens exactly is up to the implementing back-end, e.g. for an error, it could continue at the error or start from scratch again. For a finished job, it could discard all results and start from scratch again or it could just add another result and then serve both for download. This is intentionally left open as we don't know how back-ends work internally. I I'd need to come up with a best practice I would say that restarting from scratch should be possible from the three status as continuing after an error seems pretty much undefined and/or doesn't make sense with a change in the process graph.

**Comment by m-mohr** [comment]
@soxofaan What do you think? I hope it's complete. The automatic rendering is not super intuitve/well-formatted.

```mermaid
---
title: openEO Job status changes
---
stateDiagram-v2
    [*] --> created : POST /jobs
    created --> queued : POST /jobs/{job_id}/results
    queued --> running
    queued --> error
    queued --> canceled : DELETE /jobs/{job_id}/result
    note right of queued : may be skipped by some implementations
    running --> finished
    running --> error
    running --> canceled : DELETE /jobs/{job_id}/result
    canceled --> queued : may not be supported - POST /jobs/{job_id}/results
    finished --> queued : may not be supported - POST /jobs/{job_id}/results
```

**Comment by m-mohr** [comment]
Wasn't super happy with the rendering of the edges so created a Powerpoint version:

<img width="529" height="510" alt="Image" src="https://github.com/user-attachments/assets/9700aa66-d37e-4c04-b11e-83c55a2291e5" />

PS is here: #573

**Comment by soxofaan** [comment]
thanks, that is indeed informative to have

some notes:
- there is an arrow from "queued" to "created" with `DELETE /jobs/{job_id}/results`,  but that same action also causes the transition from "running" to "canceled". So with same user action you you can end up in different state depending on what current state is (queued or running). The problem is that the transition from "queued" to "running" is under control of the backend, the user has poor insights here, they can at most poll for the current status and hope they don't run in race conditions. I would always go to "canceled" from `DELETE /jobs/{job_id}/results` to keep things more predictable
- as I already discussed elsewhere (e.g. https://github.com/Open-EO/openeo-api/issues/442)  I strongly believe  it's a bad idea to allow restarting a job that is already finished, because a finished job has associated resources (job results (both data and metadata), execution logs, associated costs, ...)  so you need to specify what happens with these existing resources, and it makes things  a lot harder both at backend and client-side.
- I'm also not a fan of restarting a job in error/canceled state for the same reasons, but I can understand the appeal of still allowing that in some way


based on those thoughts I made this [quick doodle in excalidraw](https://excalidraw.com/#json=exyOh09V3QBeUUzpSXK2w,5O7brILaoOgjqO292lRJEA) of a counter proposal:

<img width="981" height="691" alt="Image" src="https://github.com/user-attachments/assets/71f58df1-2f62-402b-9efe-070e8de00be6" />

**Comment by m-mohr** [comment]
I understand your perspective and we can certainly discuss it, but this issue first of all asked to clarify what's currently in the specification. This would be solved with the diagram (I hope), it simply represents what is currently specified and documents in several places in the OpenAPI document.

I think additional changes should be at least one (or multiple) separate issue(s) for discussion, especially as strictly speaking some of the proposed changes are breaking.

I hope we can merge the diagram as a short-term solution for 1.3.0 to clarify the spec and then potentially discuss (and PR) additional changes for 1.4.0 or 2.0.0.

**Comment by soxofaan** [comment]
ok fair, then I'm fine with the diagram as is. And I'll create separate tickets for the issues mentioned above

a minor nitpick maybe: the "optionality" of the "queued" node makes things a bit confusing. I tried to clarify it a bit in my attempt by putting "created" "queued" and "running" more in a triangle configuration. But maybe it's just easier to create two versions of the diagram side by side?

**Comment by m-mohr** [comment]
Yeah, but there are other arrows pointing towards queued, which makes it a bit complex as they would also need to point towards running. I think I'd prefer it as is and just say that queued can immediately be jump to running or so.
